lib/collection/src/shards/shard_holder/resharding.rs

```diff
@@ -374,11 +374,11 @@ impl ShardHolder {
         T: Clone + MergeFilter,
     {
         let Some(state) = self.resharding_state() else {
-            return request.into();
+            return From::from(request);
```

Comment:
Why though? Does not make sense to me. Especially as `From::from` (instead of, e.g., `ReshardableRequest::from`).

Reply:
`Into` exists to make your code less annoying, use it!

Reply:
Except that it has terrible navigation

Reply:
Is rust-analyzer clever enough to jump to correct implementation of `From` when using `From::from`? 🤔